File: Clients/Node/Core/src/services/ProviderModelsService.ts
import type { FetchBasedClient } from '../client/FetchBasedClient';
import { createClientAdapter, type IFetchBasedClientAdapter } from '../client/ClientAdapter';
import type { RequestOptions } from '../client/types';

/**
 * Service for retrieving provider model information.
 */
export class ProviderModelsService {
  private readonly baseEndpoint = '/api/provider-models';
  private readonly clientAdapter: IFetchBasedClientAdapter;

  constructor(client: FetchBasedClient) {
    this.clientAdapter = createClientAdapter(client);
  }

  /**
   * Gets available models for a specified provider.
   * @param providerName - Name of the provider
   * @param forceRefresh - Whether to bypass cache and force refresh
   * @returns List of available model IDs
   */
  async getProviderModels(
    providerName: string, 
    forceRefresh = false,
    options?: RequestOptions
  ): Promise<string[]> {
    if (!providerName?.trim()) {
      throw new Error('Provider name is required');
    }

    const queryParams = forceRefresh ? '?forceRefresh=true' : '';
    
    const response = await this.clientAdapter.get<string[]>(
      `${this.baseEndpoint}/${encodeURIComponent(providerName)}${queryParams}`,
      options
    );

    return response;
  }

  /**
   * Static validation helper to validate provider name.
   */
  static validateProviderName(providerName: string): void {
    if (!providerName?.trim()) {
      throw new Error('Provider name is required');
    }
  }
}

File: archive/Clients.DotNet/ConduitLLM.CoreClient/src/Services/ProviderModelsService.cs
using ConduitLLM.CoreClient.Client;
using ConduitLLM.CoreClient.Utils;
using ConduitLLM.CoreClient.Exceptions;
using Microsoft.Extensions.Logging;

namespace ConduitLLM.CoreClient.Services
{
    /// <summary>
    /// Service for retrieving provider model information.
    /// </summary>
    public class ProviderModelsService
    {
        private readonly BaseClient _client;
        private readonly ILogger<ProviderModelsService>? _logger;

        /// <summary>
        /// Initializes a new instance of the <see cref="ProviderModelsService"/> class.
        /// </summary>
        /// <param name="client">The base client instance.</param>
        /// <param name="logger">Optional logger instance.</param>
        public ProviderModelsService(BaseClient client, ILogger<ProviderModelsService>? logger = null)
        {
            _client = client ?? throw new ArgumentNullException(nameof(client));
            _logger = logger;
        }

        /// <summary>
        /// Gets available models for a specified provider.
        /// </summary>
        /// <param name="providerName">Name of the provider.</param>
        /// <param name="forceRefresh">Whether to bypass cache and force refresh.</param>
        /// <param name="cancellationToken">Cancellation token.</param>
        /// <returns>List of available model IDs.</returns>
        public async Task<List<string>> GetProviderModelsAsync(
            string providerName, 
            bool forceRefresh = false,
            CancellationToken cancellationToken = default)
        {
            if (string.IsNullOrWhiteSpace(providerName))
            {
                throw new ArgumentException("Provider name is required", nameof(providerName));
            }

            try
            {
                _logger?.LogDebug("Getting models for provider {ProviderName} (forceRefresh: {ForceRefresh})", providerName, forceRefresh);
                
                var url = $"/api/provider-models/{Uri.EscapeDataString(providerName)}";
                if (forceRefresh)
                {
                    url += "?forceRefresh=true";
                }

                var response = await _client.GetForServiceAsync<List<string>>(url, cancellationToken: cancellationToken);
                _logger?.LogDebug("Retrieved {ModelCount} models for provider {ProviderName}", response?.Count ?? 0, providerName);
                return response ?? new List<string>();
            }
            catch (Exception ex) when (!(ex is ConduitCoreException))
            {
                ErrorHandler.HandleException(ex);
                throw;
            }
        }
    }
}